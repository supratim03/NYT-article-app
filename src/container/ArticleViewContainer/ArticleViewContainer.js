import React from "react";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import moment from "moment";
import "../../styles/ArticleViewContainer.css";
import back from "../../assets/left-arrow.png";
import right from "../../assets/right.png";

const ArticleViewContainer = ({ article, searchKey }) => {
	const navigate = useNavigate();

	const handleArticleView = url => {
		window.open(url, "_blank");
	};

	const backToSearch = () => {
		navigate(`/search?query=${searchKey}`);
	};
	return (
		<div className="background-wrapper">
			<div className="article-view-wrapper">
				<div>
					<span 
						className="sa-link sa-cursor-pointer back" 
						onClick={backToSearch}
					>
						<span className="vertical-center">
							<img src={back} alt="back" style={{height: "12px"}}/>
						</span>&nbsp;
						Back to Search Result
					</span>
				</div>
				<div className="sa-padding-top-20">
					<span className="sa-fw-b sa-fs-18">{article?.abstract}</span>
				</div>
				<div className="sa-padding-top-20">
					<span>{moment(article.pub_date).format("DD MMM, YYYY")}</span>
				</div>
				<div className="article-desc">
					<p>
						{article.lead_paragraph}
					</p>
				</div>
				<div>
					<span 
						className="sa-link sa-cursor-pointer articleView" 
						onClick={() => handleArticleView(article.web_url)}
					>
						Read the full article
						<span className="vertical-center">
							<img src={right} alt="right" style={{height: "16px"}} />
						</span>
					</span>
				</div>
			</div>
		</div>
	);
};

const mapStateToProps = state => {
	const { SearchReducer } = state;
	return {
		article: SearchReducer.selectedArticle,
		searchKey: SearchReducer.searchKey
	};
};

export default connect(mapStateToProps, {})(ArticleViewContainer);
