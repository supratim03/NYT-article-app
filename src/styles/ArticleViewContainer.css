.article-view-wrapper {
    padding: 200px !important;
}

.article-desc {
    width: 75%;
}

.background-wrapper {
    background: url("../assets/Background.png");
    height: 100vh;
}

.vertical-center {
    vertical-align: middle;
}

@media only screen and (max-width: 1000px) {
    .article-view-wrapper {
        padding-left: 20px !important;
        padding-right: 20px !important;
        padding-top: 150px !important;
    }

    .article-desc {
        width: 100% !important;
    }
}