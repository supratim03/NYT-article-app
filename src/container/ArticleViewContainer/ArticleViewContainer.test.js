import React from "react";
import { mount } from "enzyme/build";
import configureStore from "redux-mock-store";
import createSagaMiddleware from "@redux-saga/core";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import ArticleViewContainer from "./ArticleViewContainer";

// mock useNavigate
const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
   useNavigate: () => mockedUsedNavigate
 }));

describe("ArticleView container", () => {

    const sagaMiddleWare = createSagaMiddleware();
    const mockStore = configureStore([sagaMiddleWare]);

    const initialState = {
        "SearchReducer": {
            "searchKey": "USA",
            "selectedArticle": {
                "abstract": "test abstract",
                "pub_date": "2022-05-06T09:00:11+0000",
                "lead-paragraph": "test lead paragraph",
                "web_url": "https://www.nytimes.com/2022/05/06/opinion/ezra-klein-podcast-matt-continetti.html",
                "_id": 1
            }
        }
    };

    const store = mockStore(initialState);
    let wrapper;

    beforeEach(() => {
        const componentProps = {
            "article" : initialState.SearchReducer.selectedArticle,
            "searchKey": initialState.SearchReducer.searchKey
        };

        wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <ArticleViewContainer {...componentProps} />
                </MemoryRouter>
            </Provider>
        );
    });

    it("render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("open new window", () => {
        window.open = jest.fn();
        const element = wrapper.find(".articleView").at(0);
        element.simulate("click");
        expect(window.open).toHaveBeenCalledTimes(1);
    });

    it("navigate to search page", () => {
        const element = wrapper.find(".back").at(0);
        element.simulate("click");
        expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
    });
});

