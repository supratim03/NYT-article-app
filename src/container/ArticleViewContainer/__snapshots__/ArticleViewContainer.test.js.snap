// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleView container render correctly 1`] = `ReactWrapper {}`;
